os/net/socket/net_sockets.c

```diff
@@ -133,10 +136,22 @@ void net_initlist(FAR struct socketlist *list)
 
 void net_releaselist(FAR struct socketlist *list)
 {
-	DEBUGASSERT(list);
+	int i;
+	int ret;
+	struct lwip_sock *psock;
+	for (i = 0; i < CONFIG_NSOCKET_DESCRIPTORS; ++i) {
+		if (list->sl_sockets[i].sock) {
+			psock = (struct lwip_sock *)list->sl_sockets[i].sock;
+			DEBUGASSERT(psock->conn);
```

Comment:
Why did you check `psock->conn` is NULL or not? `psock->conn` is not used anywhere in this function.

Reply:
because if socket is there but connection is not there then this is a problem.

Reply:
In that case, I think we'd better check it when it is used.

Reply:
I didn't get your point clearly. Here, i am checking socket conn because i am fetching socket descriptor from netconn and with that SD i am able assign NULL to corresponding socket after releasing the socket resources.